diesel/src/associations/belongs_to.rs

```diff
@@ -27,8 +27,9 @@ impl<Parent, Child, Iter> GroupedBy<Parent> for Iter where
         let id_indices: HashMap<_, _> = parents.iter().enumerate().map(|(i, u)| (u.id(), i)).collect();
```

Comment:
I mean the keys aren't necessarily always small. String keys are a thing and the user can do literally anything here. That said, I'm open to using a different hasher if there's a good reason. I'm not sure anything can end up being a security risk here, the worst it could be is a DOS vector since collisions would mean slower execution.